store/sqlstore/channel_store.go

```diff
@@ -3640,23 +3656,18 @@ func (s SqlChannelStore) UpdateSidebarCategories(userId, teamId string, categori
 	for _, categoryToUpdate := range categories {
 		categoryWithChannels, appErr := s.GetSidebarCategory(categoryToUpdate.Id)
 		if appErr != nil {
-			return nil, model.NewAppError("SqlPostStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, err.Error(), http.StatusInternalServerError)
+			return nil, model.NewAppError("SqlPostStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.app_error", nil, appErr.Error(), http.StatusInternalServerError)
 		}
+
 		category := &categoryWithChannels.SidebarCategory
+
 		// we can update display name only on custom categories
 		if category.DisplayName != categoryToUpdate.DisplayName {
 			if category.Type != model.SidebarCategoryCustom {
-				return nil, model.NewAppError("SqlPostStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.invalid_update_fields", map[string]interface{}{"field": "DisplayName"}, err.Error(), http.StatusBadRequest)
+				return nil, model.NewAppError("SqlPostStore.UpdateSidebarCategory", "store.sql_channel.sidebar_categories.invalid_update_fields", map[string]interface{}{"field": "DisplayName"}, "", http.StatusBadRequest)
 			}
 			category.DisplayName = categoryToUpdate.DisplayName
 		}
-		// we can update sort method only on DM category
```

Comment:
was this requirement changed?

Reply:
I should've changed it with my manual vs default sort method PR. I'm setting the sort order to manual when someone rearranges their channels manually. It could make it weird if someone uses the API to set their sort order to something different using the API, but I think everything should still work fine regardless. Sorry about making you add it when I just ended up removing it.